lib/rules/no-array-prototype-extensions.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const { getImportIdentifier } = require('../utils/import');
+
+const ERROR_MESSAGE = "Don't use Ember's array prototype extensions";
+
+const EXTENSION_METHODS = new Set([
+  /** EmberArray */
+  'any',
+  'compact',
+  'filterBy',
+  'findBy',
+  'getEach',
+  'invoke',
+  'isAny',
+  'isEvery',
+  'mapBy',
+  'objectAt',
+  'objectsAt',
+  'reject',
+  'rejectBy',
+  'setEach',
+  'sortBy',
+  'toArray',
+  'uniq',
+  'uniqBy',
+  'without',
+  /** MutableArray */
+  'addObject',
+  'addObjects',
+  'clear',
+  'insertAt',
+  'popObject',
+  'pushObject',
+  'pushObjects',
+  'removeAt',
+  'removeObject',
+  'removeObjects',
+  'reverseObjects',
+  'setObject',
+  'shiftObject',
+  'unshiftObject',
+  'unshiftObjects',
+]);
+
+const EXTENSION_PROPERTIES = new Set(['lastObject', 'firstObject']);
+//----------------------------------------------------------------------------------------------
+// General rule - Don't use Ember's array prototype extensions like .any(), .pushObject() or .firstObject
+//----------------------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: "disallow usage of Ember's `Array` prototype extensions",
+      category: 'Deprecations',
+      recommended: false,
+      url: 'https://github.com/ember-cli/eslint-plugin-ember/tree/master/docs/rules/no-array-prototype-extensions.md',
+    },
+    fixable: null,
+    schema: [],
+  },
+
+  ERROR_MESSAGE,
+
+  create(context) {
+    let importedEmberName;
+
+    return {
+      CallExpression(node) {
```

Comment:
Can you add a comment at the top of each of these visitor functions explaining what it's checking for with a code sample?